app.ts

```diff
@@ -0,0 +1,391 @@
+const globals = globalInitializer();
+let isFunctionRunning = false;
+
+// The following would handle all the variable properties that is being changed a lot
+function globalInitializer() {
+	return {
+		api: "http://localhost:2050/",
+		currentPage: 1,
+		firstPage: 1,
+		lastPage: 10,
+		currentFromID: 1,
+		currentToID: 20,
+		difference: 0,
+	};
+};
+
+// This function  will handle retrieving the records from the api
+async function getRecords(fromID: number, toID: number): Promise<string[][]> {
+	try {
+		const response = await fetch(`${globals.api}records?from=${fromID}&to=${toID}`);
+		const data = await response.text();
+		const records = JSON.parse(data);
```

Comment:
```suggestion
		const records = JSON.parse(response.text());
```

Reply:
@DevinFledermaus... you can't parse a promise.